core/lib/libtomcrypt/mac.c

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2014-2019, Linaro Limited
+ */
+
+#include <assert.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <tomcrypt.h>
+#include <utee_defines.h>
+#include <util.h>
+
+/*****************************************************************************
+ * Message Authentication Code functions
+ *****************************************************************************/
+
+#if defined(CFG_CRYPTO_HMAC)
```

Comment:
We could save ifdefs by splitting into two files:
```
srcs-$(CFG_CRYPTO_HMAC) += hmac.c 
srcs-$(CFG_CRYPTO_OMAC) += omac.c
```
This is cleaner IMO but I think you prefer otherwise? ;-)
It does not matter much anyway.

Reply:
Agree, I'll fix.